src/stepfunctions/steps/service.py

```diff
@@ -887,3 +892,59 @@ def __init__(self, state_id, **kwargs):
+            comment (str, optional): Human-readable comment or description. (default: None)
+            input_path (str, optional): Path applied to the state’s raw input to select some or all of it; that selection is used by the state. (default: '$')
+            parameters (dict, optional): The value of this field becomes the effective input for the state.
+            result_path (str, optional): Path specifying the raw input’s combination with or replacement by the state’s result. (default: '$')
+            output_path (str, optional): Path applied to the state’s output after the application of `result_path`, producing the effective output which serves as the raw input for the next state. (default: '$')
+            wait_for_completion (bool, optional): Boolean value set to `True` if the Task state should wait to complete before proceeding to the next step in the workflow. (default: True)
+            json_output(bool, optional): Boolean value set to `True` if the the Task should return a response in JSON format. If set to `False`, the Task should return a str response (default: False)
+        """
+        if wait_for_callback:
+            """
+            Example resource arn: arn:aws:states:::states:startExecution.waitForTaskToken
+            """
+
+            kwargs[Field.Resource.value] = get_service_integration_arn(STEP_FUNCTIONS_SERVICE_NAME,
+                                                                       StepFunctions.StartExecution,
+                                                                       IntegrationPattern.WaitForTaskToken)
+        elif wait_for_completion:
+            if json_ouput:
+                """
+                Example resource arn:aws:states:::states:startExecution.sync:2
+                """
+                kwargs[Field.Resource.value] = get_service_integration_arn(STEP_FUNCTIONS_SERVICE_NAME,
+                                                                       StepFunctions.StartExecution,
+                                                                       IntegrationPattern.WaitForCompletionWithJsonResponse)
+            else:
+                """
+                Example resource arn:aws:states:::states:startExecution.sync
+                """
+                kwargs[Field.Resource.value] = get_service_integration_arn(STEP_FUNCTIONS_SERVICE_NAME,
+                                                                       StepFunctions.StartExecution,
+                                                                       IntegrationPattern.WaitForCompletion)
+        else:
+            """
+            Example resource arn:aws:states:::states:startExecution
+            """
+            kwargs[Field.Resource.value] = get_service_integration_arn(STEP_FUNCTIONS_SERVICE_NAME,
+                                                                       StepFunctions.StartExecution)
```

Comment:
suggestion: should there be a flag to ask for this? 

Currently, Asking for this property is also counter-intuitive because it's a behaviour that is achieved by setting `wait_for_callback` and `wait_for_completion` to false. 

A customer saying "no" to these 2 properties != a "yes" for the behaviour they will get. Additionally, this behaviour is not currently documented in any way.

Recommendation:
* Add a flag to opt into the behaviour. Clearly document what happens when that flag is set to `True`
  *   Alternatively could model these 3 properties in a more complex type, but I'm not leaning that way because that might be disruptive and too different from usage of other service integrations.

Reply:
I like your suggestion - I agree, having to disable to properties to activate another is not intuitive. A flag might make things clearer!
I had thought of introducing an enum for the property type, but as you mentioned, the implementation would diverge from the other service integrations.
I can add a check to see if more than 1 is enabled and throw an exception if that is the case.

Reply:
What would be the default if all flags are disabled? Should we return an error?
If we were to keep only the 2 initial property flags `wait_for_callback` and `wait_for_completion` and:
- Raise an exception if both were set to True (as you mentioned below) 
- Update documentation to illustrate how to select this behaviour (Disable both false to get arn:aws:states:::states:startExecution resource)

I think this way, would be clearer for the customers instead of having them enable at least 1 of the 3 flags - What do you think?

Reply:
> If we were to keep only the 2 initial property flags wait_for_callback and wait_for_completion

How would users use the async invocation method, this only enables 2/3 possible modes of invoking the service integration. I think we need to provide users with an intuitive way to ask for `arn:aws:states:::states:startExecution`.

I'm not in favour of setting `wait_for_callback` and `wait_for_completion` to false to achieve that. It's counter-intuitive to me as a developers, but perhaps others will disagree...


* the default if none of the properties are set should be to use `wait_for_completion` aka `sync:2`
  * Rationale: it's what all the other service integrations establish as a default
* if a user sets all the flags to False, then it's a validation error
  * Rationale: it's required that one of the modes of operation is established and the user's intention is unclear.

Aside:
These implementation decisions should be captured and elaborated in the PR summary itself. Right now the description only captures the outcome, but it's valuable context for posterity as well as helping reviewers understand why the code is structured the way it is. Please ensure you update the PR description with implementation decisions that are being made.

Reply:
>- if a user sets all the flags to False, then it's a validation error
Rationale: it's required that one of the modes of operation is established and the user's intention is unclear.

Agreed - this will ensure the customer's intention is clear. 
I'll go with what you are proposing - we want to provide an intuitive solution for our customers

Reply:
>Please ensure you update the PR description with implementation decisions that are being made.

Updated the description